app/services/catalog/update_order_item.rb

```diff
@@ -75,6 +75,7 @@ def mark_item_failed
       @order_item.completed_at = DateTime.now
```

Comment:
Not sure if we want to do it in this PR, but I see a huge opportunity to DRY-UP this methods and `mark_item_finished`.

Reply:
I 100% agree, there is a ticket out there that @syncrou and I created to add methods on the OrderItem model to do this, since we do it all over the app. Maybe I'll add it in another commit here quick.

Reply:
I'm good either way.  Rather get it fixed in sooner and refactor if we think it will add much time to merged the PR.  Meaning, will miss this sprint.

Reply:
After chewing on it for a bit I'm going to do a follow-up PR with those changes, heres the ticket I referred to earlier to track: https://projects.engineering.redhat.com/browse/SSP-1265